src/image.rs

```diff
@@ -1001,8 +1002,35 @@ impl ImageBuildChunk {
         }
         None
     }
+
+    /// Returns the image layer ID and size during download.
+    ///
+    /// If both the layer ID and eventual (compressed) layer size are available
+    /// from the ImageBuildChunk, they will be returned as `Some((layer_id,
+    /// layer_size))`. Otherwise, `None` is returned.
+    pub fn image_layer_bytes(&self) -> Option<(String, u64)> {
+        match self {
+            ImageBuildChunk::PullStatus {
+                status: _,
+                id,
+                progress: _,
+                progress_detail,
+            } => {
+                if let Some(id) = id {
```

Comment:
Nested `if let` can be removed and their logic moved into the `match` expression above using https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#destructuring-nested-structs-and-enums. This will also remove unnecessary `return` below and make your function consist of only a single `match` expression